airflow/www/utils.py

```diff
@@ -812,6 +819,56 @@ def get_col_default(self, col_name: str) -> Any:
     filter_converter_class = AirflowFilterConverter
 
 
+class DagRunCustomSQLAInterface(CustomSQLAInterface):
+    """
+    Overwrite of custom delete and delete_all methods for speeding up
+    the DagRun deletion when DagRun has a lot of task instances.
+    The default cascade deletion was performing very slowly when task instances were more than 10k.
+
+    """
+
+    def delete(self, item: Model, raise_exception: bool = False) -> bool:
+        try:
+            self.session.query(TaskInstance).where(TaskInstance.run_id == item.run_id).delete()
+            return super().delete(item, raise_exception=raise_exception)
+        except IntegrityError as e:
+            self.message = (as_unicode(self.delete_integrity_error_message), "warning")
+            log.warning(LOGMSG_WAR_DBI_DEL_INTEGRITY.format(str(e)))
+            self.session.rollback()
+            if raise_exception:
+                raise e
+            return False
+        except Exception as e:
+            self.message = (
+                as_unicode(self.general_error_message + " " + str(sys.exc_info()[0])),
+                "danger",
+            )
+            log.exception(LOGMSG_ERR_DBI_DEL_GENERIC.format(str(e)))
+            self.session.rollback()
+            if raise_exception:
+                raise e
+            return False
```

Comment:
```suggestion
        self.session.query(TaskInstance).where(TaskInstance.run_id == item.run_id).delete()
        return super().delete(item, raise_exception=raise_exception)
```

I’m not 100% sure, but since `super().delete` contains some re-raising, duplicating the exception blocks here would cause an error to be logged twice (one in `super()` and one here). And since those errors can only happen when you call `commit()` (in `super()`), error handling here is never reached.

Reply:
Make sense, removed the redundant exception handling.